src/Components/AddIssue/index.tsx

```diff
@@ -1,71 +1,139 @@
     {
       super(props);
-
+      this._issueLoadLocationGateway = context.get<ILoadIssueLocationGateway>("ILoadIssueLocationGateway");
+      this.loadIssueLocations();
       this.state={
-        issue: this.props.issue
+        issue: this.props.issue,
+        issueTypes: this._issueTypes,
+        issueLocations: this._issueLocations,
       }
     }
-    deleteIssue(){
+
+    loadIssueLocations() {
+      this._issueLoadLocationGateway.loadIssueLocations().then((data) => {
+        this._issueLocations = data.issueLocations;
+        this.setState({ issueLocations: data.issueLocations });
+        let issue = this.state.issue;
+        issue.Location = this.state.issueLocations[0];
+        this.setState({ issue: issue });
+      });
+    }
+
+    deleteIssue = (): void => {
       this.props.onDeleteIssue(this.props.index);
     }
 
-    handleChangeOfIssue = (event: ChangeEvent<HTMLInputElement>): void => {
-      const name = event.target.name;
+    handleChangeOfIssueNote=(event: ChangeEvent<HTMLTextAreaElement>): void => {
       const value = event.target.value;
-      this.setState({
-          issue: { 
-              ...this.state.issue,
-              [name]: [value]
-          }
-      });
+
+      let issue = this.state.issue;
+      issue.Notes = value;
+      this.setState({issue:issue});
+
+      this.props.onChangeIssue(this.state.issue, this.props.index);
+      console.log(this.state.issue);
+    }
+
+    handleChangeOfIssueTypeDropDownList=(event:React.ChangeEvent<HTMLSelectElement>): void => {
+      console.log("handleChangeOfIssueTypeDropDownList");
       
-      this.props.onChangeIssue(this.state.issue, this.props.index)
+      const index = Number(event.target.value);
+      let issueType = this.state.issueTypes[index];
+      let issue = this.state.issue;
+      issue.IssueType = issueType;
+      this.setState({issue:issue});
+
+      this.props.onChangeIssue(this.state.issue, this.props.index);
+      console.log(issue);
     }
 
-    handleChangeOfIssueNote=(event: ChangeEvent<HTMLTextAreaElement>): void => {
-      const value = event.target.value;
-      this.setState({
-        issue: { 
-            ...this.state.issue,
-            Notes: value
-        }
-      });
+    handleChangeOfIssueLocationDropDownList=(event:React.ChangeEvent<HTMLSelectElement>): void => {
+      console.log("handleChangeOfIssueLocationDropDownList");
+      
+      const index = Number(event.target.value);
+      let location = this.state.issueLocations[index];
+
+      let issue = this.state.issue;
+      issue.Location = location;
+      this.setState({issue:issue });
 
-      this.props.onChangeIssue(this.state.issue, this.props.index)
+      this.props.onChangeIssue(this.state.issue, this.props.index);
+      console.log(this.state.issue);
     }
 
+    createSelectItemsForIssueTypes() {
+      let items = [];         
+      for (let i = 0; i < this.state.issueTypes.length ; i++) {
```

Comment:
I really think these would be cleaner as map statements but this works so not a big deal

Reply:
Ok, will change that

Reply:
Resolved in https://github.com/LBHackney-IT/lbh-etra-front-end/pull/14/commits/7eae314f2f10a1808c5446a1e12922d269af4536